### ISSUE TITLE
[data grid] Pinned column breaks JAWS accessibility

### ISSUE THREAD

**Issue by bmayen** [mui/mui-x#5568]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When using pinned columns *and JAWS keyboard navigation*, items in pinned columns are read as "end of row" and user cannot navigate by JAWS keyboard commands to next column. This appears to be because cells in a pinned column are grouped under a separate 'MuiDataGrid-pinnedColumns' div which jaws interprets as the entire table and doesn't see the subsequent 'MuiDataGrid-virtualScrollerRenderZone' div containing the rest of the table's cells.

### Expected behavior 🤔

Pinned columns should be treated no differently than regular columns.

### Steps to reproduce 🕹

Link to live example:

Steps:
1.
2.
3.
4.

### Context 🔦

Looking for JAWS compatibility

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Chrome

System:
    OS: macOS 12.4
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v16.13.1/bin/yarn
    npm: 8.13.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Not Found
    Firefox: 101.0
    Safari: 15.5
  npmPackages:
    @emotion/react: ^11.8.1 => 11.9.0
    @emotion/styled: ^11.8.1 => 11.8.1
    @mui/base:  5.0.0-alpha.75
    @mui/icons-material: ^5.6.0 => 5.6.0
    @mui/material: ^5.6.0 => 5.6.0
    @mui/private-theming:  5.8.0
    @mui/styled-engine:  5.6.0
    @mui/styles: ^5.6.2 => 5.8.0
    @mui/system:  5.6.0
    @mui/types:  7.1.3
    @mui/utils:  5.8.0
    @mui/x-data-grid:  5.11.1
    @mui/x-data-grid-pro: 5.11.1 => 5.11.1
    @mui/x-date-pickers: ^5.0.0-alpha.5 => 5.0.0-alpha.5
    @mui/x-license-pro:  5.11.1
    @types/react: ^17.0.38 => 17.0.39
    react: ^17.0.2 => 17.0.2
    react-dom: ^17.0.2 => 17.0.2
    typescript: ^4.5.4 => 4.6.2
```
</details>

**Comment by alexfauquette** [comment]
I'm wondering if this issue can be solved with [aria-posinset](https://w3c.github.io/aria/#aria-posinset) and [aria-setsize](https://w3c.github.io/aria/#aria-setsize)

**Comment by alexfauquette** [comment]
@m4theushw I tries to reproduce this issue with JAWS on browser stack. But I never get a message about "end of rows". Do you know how to reproduce it?

**Comment by alexfauquette** [comment]
That's the right place to add your comment 😉 Thanks for the feedback

**Comment by dlee3** [comment]
The company I work for has a MUI X Pro license and we have been experiencing this same issue. We have around 10 tables in our application and we have a contract where we must have full JAWS accessibility.

If there are no pinned columns then the JAWS navigation works without issue. But when there are any pinned columns the navigation will only navigate the pinned columns before going to the next row.
ie - If you have 20 columns total and 2 pinned columns, it will navigate the 2 pinned columns for every row but you cannot access the other columns without a mouse click which is not something a normal JAWS user could do.

Here's our order information:
[Order #48348] (August 1, 2022)

**Comment by gportela85** [comment]
@alexfauquette this has been here for a long time and while I was attempting to find a workaround around for it, I noticed that this is only a problem with Chrome. It works fine with Firefox so I filed this issue with Chromium. If this is still an issue for you guys, might be worth adding a +1 there. https://issues.chromium.org/issues/479290178

**Comment by cherniavskii** [comment]
I believe this issue is solved by https://github.com/mui/mui-x/pull/12376 since pinned cells are rendered within the same row DOM node
